numba/typing/arraydecl.py

```diff
@@ -612,22 +612,38 @@ def sum_expand(self, args, kws):
     sum can be called with or without an axis parameter.
     """
     pysig = None
-    if kws:
+    if 'axis' in kws and not 'dtype' in kws:
         def sum_stub(axis):
             pass
         pysig = utils.pysignature(sum_stub)
         # rewrite args
         args = list(args) + [kws['axis']]
-        kws = None
+        # kws = None
```

Comment:
dead code?

Reply:
tbh, I couldn't understand why that was there so left it in case I had to restore it. Since everything is working I can delete it. However, it's that feeling you get when you put together a machine and there's a piece left outside. I'll delete it.